CHANGES.md

```diff
@@ -30,6 +30,11 @@ OpenSSL 3.0
 
    *Shane Lontis*
 
+ * The SSL option SSL_OP_IGNORE_UNEXPECTED_EOF restoring old behaviour when peer
+   does not send close_notify is introduced.
```

Comment:
The behaviour you get is not the old behaviour.

Reply:
The current implementation probably does what you document, but it's not what was discussed. If that option is set, the unexpected EOF is ignored, it pretends close notify was received instead and so returns SSL_ERROR_ZERO_RETURN.

Reply:
Does it mean we still need to update the code? Sorry, I don't fill familiar enough with it.

Reply:
Yes, the code needs to change. I think at least s->shutdown needs to set SSL_RECEIVED_SHUTDOWN and s->s3.warn_alert needs to be set to SSL_AD_CLOSE_NOTIFY